src/vs/workbench/contrib/terminal/browser/terminalTabbedView.ts

```diff
@@ -119,6 +120,20 @@ export class TerminalTabbedView extends Disposable {
 				}
 			}
 		}));
+		this._register(this._terminalService.onDidChangeActiveInstance(() => {
+			const terminalScopedContextKeyService = this._terminalService.activeInstance?.scopedContextKeyService;
+
+			this._instanceMenu.dispose();
+			this._instanceMenu = this._register(menuService.createMenu(
+				MenuId.TerminalInstanceContext,
+				terminalScopedContextKeyService || contextKeyService
+			));
```

Comment:
Using a `MutableDisposable` is the best pattern for this type of thing. That way you don't need to worry about calling `dispose()` at all.